File: src/monitor/tracker.py
from datetime import datetime

class AuctionMonitor:
    """Real-time auction monitoring"""

    def __init__(self, database):
        self.db = database
        self.watchlist = []
        self.alerts = []

    def add_to_watchlist(self, auction_id, options=None):
        """Add auction to watch list"""
        options = options or {}

        self.watchlist.append({
            'auction_id': auction_id,
            'target_price': options.get('target_price'),
            'max_bid': options.get('max_bid'),
            'tracked_bidders': options.get('tracked_bidders', []),
            'added_at': datetime.now()
        })
        print(f'📍 Added auction {auction_id} to watchlist')

    def check_auction(self, auction_id, current_data):
        """Check for changes in auction"""
        conn = self.db.get_connection()
        cursor = conn.cursor()

        # Get previous data
        cursor.execute('SELECT current_bid, bid_count FROM auctions WHERE ba_id = ?', (auction_id,))
        result = cursor.fetchone()

        if result:
            previous_bid, previous_count = result

            # Detect price movement
            if current_data['current_bid'] > previous_bid:
                self.alerts.append({
                    'type': 'PRICE_INCREASE',
                    'auction_id': auction_id,
                    'old_bid': previous_bid,
                    'new_bid': current_data['current_bid'],
                    'increase': current_data['current_bid'] - previous_bid
                })

            # Detect new bids
            if current_data['bid_count'] > previous_count:
                self.alerts.append({
                    'type': 'NEW_BID',
                    'auction_id': auction_id,
                    'bid_count': current_data['bid_count']
                })

        conn.close()

    def get_alerts(self):
        """Get all alerts"""
        return self.alerts

    def clear_alerts(self):
        """Clear alerts"""
        self.alerts = []

File: src/monitor/tracker.js
// Real-time auction monitoring
// Tracks price changes, bid activity, time remaining

class AuctionMonitor {
  constructor(database) {
    this.db = database;
    this.watchlist = [];
    this.alerts = [];
  }

  async addToWatchlist(auctionId, options = {}) {
    this.watchlist.push({
      auctionId,
      targetPrice: options.targetPrice,
      maxBid: options.maxBid,
      trackedBidders: options.trackedBidders || [],
      addedAt: new Date()
    });
    console.log(`📍 Added auction ${auctionId} to watchlist`);
  }

  async checkAuction(auctionId, currentData) {
    const previous = await this.db.getAuctionData(auctionId);

    // Detect price movement
    if (previous && currentData.currentBid > previous.currentBid) {
      this.alerts.push({
        type: 'PRICE_INCREASE',
        auctionId,
        oldBid: previous.currentBid,
        newBid: currentData.currentBid,
        increase: currentData.currentBid - previous.currentBid
      });
    }

    // Detect new bidders
    if (currentData.bidCount > previous.bidCount) {
      this.alerts.push({
        type: 'NEW_BID',
        auctionId,
        bidCount: currentData.bidCount
      });
    }

    // Check if tracked competitors are bidding
    if (currentData.lastBidder && this.isTrackedCompetitor(currentData.lastBidder)) {
      this.alerts.push({
        type: 'COMPETITOR_ACTIVE',
        auctionId,
        competitor: currentData.lastBidder
      });
    }

    await this.db.updateAuction(auctionId, currentData);
  }

  isTrackedCompetitor(username) {
    // Check against list of competitors to monitor
    return false; // TODO: implement
  }

  getAlerts() {
    return this.alerts;
  }

  clearAlerts() {
    this.alerts = [];
  }
}

module.exports = AuctionMonitor;
